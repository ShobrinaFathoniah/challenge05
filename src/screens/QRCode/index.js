import {View, Alert} from 'react-native';
import React from 'react';
import {CameraScreen, CameraType} from 'react-native-camera-kit';
// import {useIsFocused} from '@react-navigation/native';

const QRCode = () => {
  // const isFocused = useIsFocused();
  const onReadCode = data => {
    Alert.alert('Notification', data.nativeEvent.codeStringValue);
  };
  return (
    <View>
      <CameraScreen
        cameraType={CameraType.Back}
        scanBarcode={true}
        onReadCode={event => onReadCode(event)} // optional
        showFrame={true} // (default false) optional, show frame with transparent layer (qr code or barcode will be read on this area ONLY), start animation for scanner,that stoped when find any code. Frame always at center of the screen
        laserColor="red" // (default red) optional, color of laser in scanner frame
        frameColor="white" // (default white) optional, color of border of scanner frame
      />
    </View>
  );
};

export default QRCode;
