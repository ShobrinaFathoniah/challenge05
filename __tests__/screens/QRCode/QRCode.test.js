import React from 'react';
import QRCode from '../../../src/screens/QRCode';
import {create} from 'react-test-renderer';

describe('Test Snapshoot', () => {
  test('QRCode Snapshoot', () => {
    jest.useFakeTimers();

    const snap = create(<QRCode />);
    expect(snap).toMatchSnapshot();
  });
});
