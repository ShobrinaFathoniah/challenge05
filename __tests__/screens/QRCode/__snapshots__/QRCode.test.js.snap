// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Snapshoot QRCode Snapshoot 1`] = `
<View>
  <View
    allowCaptureRetake={false}
    cameraType="back"
    frameColor="white"
    laserColor="red"
    onReadCode={[Function]}
    saveToCameraRoll={true}
    scanBarcode={true}
    showFrame={true}
    style={
      Object {
        "backgroundColor": "black",
        "flex": 1,
      }
    }
  >
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 0,
          "paddingTop": 8,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "paddingHorizontal": 15,
          }
        }
      >
        <Image
          resizeMode="contain"
          style={
            Array [
              Object {
                "flex": 1,
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "paddingHorizontal": 15,
          }
        }
      >
        <Image
          resizeMode="contain"
          style={
            Array [
              Object {
                "flex": 1,
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        />
      </View>
    </RCTSafeAreaView>
    <View
      style={
        Object {
          "flex": 10,
          "flexDirection": "column",
        }
      }
    >
      <CKCamera
        cameraOptions={
          Object {
            "ratioOverlayColor": undefined,
          }
        }
        cameraType="back"
        flashMode="auto"
        frameColor="white"
        laserColor="red"
        onReadCode={[Function]}
        resetFocusTimeout={0}
        resetFocusWhenMotionDetected={true}
        saveToCameraRoll={true}
        scanBarcode={true}
        showFrame={true}
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
          }
        }
        torchMode="off"
      />
    </View>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Array [
          Object {
            "flex": 2,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "padding": 14,
          },
          Object {
            "backgroundColor": "#ffffff00",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 20,
            }
          }
        />
      </View>
    </RCTSafeAreaView>
  </View>
</View>
`;
